src/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs

```diff
@@ -68,35 +66,12 @@ public override int WriteTimeout
             }
         }
 
-        public override bool CanRead
+        static void CheckBaudRate(int baudRate)
```

Comment:
Nit: private.  Also, is this something that can be shared with Windows, and thus moved into the shared file?